File: 백준/Silver/10773. 제로/제로.py
import sys
stack=[]
n=int(input())
answer=0
for i in range(n):
    num=int(sys.stdin.readline())
    if(num!=0):
        stack.append(num)
    else:
        stack.pop()
while(stack):
    answer+=stack.pop()
print(answer)

File: 백준/Silver/10773. 제로/제로.cc
#include <stdio.h>
#include <iostream>
#include <string>
#include <stack>
int main(){
    std::cin.tie(0);
    std::ios::sync_with_stdio(0);
    std::stack <int> stack ={};
    int n{},num{};
    std::cin >> n ;
    for(int i=0; i<n; i++){
        std::cin >> num;
        if(num!=0){
            stack.push(num);
        }
        else{
            stack.pop();
        }
    }
    int answer{};
    while(!stack.empty()){
        answer+=stack.top();
        stack.pop();
    }
    std::cout << answer;
}